src/ServiceJobList.c

```diff
@@ -0,0 +1,143 @@
+//ServiceJobList.c
+
+// SPDX-FileCopyrightText: 2024, Yaskawa America, Inc.
+// SPDX-FileCopyrightText: 2024, Delft University of Technology
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "MotoROS.h"
+
+rcl_service_t g_serviceListJobs;
+
+ServiceListJobs_Messages g_messages_ListJobs;
+
+typedef motoros2_interfaces__srv__ListJobs_Response ListJobsResponse;
+
+
+static const size_t RESP_NAMES_LIST_INITIAL_SIZE = 64;
+
+
+void Ros_ServiceListJobs_Initialize()
+{
+    MOTOROS2_MEM_TRACE_START(svc_list_jobs_init);
+
+    const rosidl_service_type_support_t* type_support =
+        ROSIDL_GET_SRV_TYPE_SUPPORT(motoros2_interfaces, srv, ListJobs);
+
+    rcl_ret_t ret = rclc_service_init_default(
+        &g_serviceListJobs, &g_microRosNodeInfo.node, type_support, SERVICE_NAME_LIST_JOBS);
+    motoRosAssert_withMsg(ret == RCL_RET_OK,
+        SUBCODE_FAIL_INIT_SERVICE_LIST_JOBS, "Failed to init service (%d)", (int)ret);
+
+    rosidl_runtime_c__String__init(&g_messages_ListJobs.response.message);
+    //NOTE: if/when needed, this will be resized
+    rosidl_runtime_c__String__Sequence__init(
+        &g_messages_ListJobs.response.names, RESP_NAMES_LIST_INITIAL_SIZE);
```

Comment:
The idea is to initialise the response with the default/initial list size and resize later in the service callback if/when needed.